imsim/stamp.py

```diff
@@ -434,14 +436,16 @@ def buildPSF(self, config, base, gsparams, logger):
                 flux = self.gal.flux*self.vignetting.at_sky_coord(
                     base['sky_pos'], self.image.wcs, pix_to_fp)
                 self.realized_flux = galsim.PoissonDeviate(self.rng, mean=flux)()
+                base['realized_flux'] = self.realized_flux
```

Comment:
I think this implies that realized_flux means two different things for vignetted objects depending on whether they were rendered with fft or not.  That's probably not ideal for an output catalog quantity.  I think we probably just want the other realized_flux always for output to be the number of photons from the object, before any possible loss from vignetting.
If you want to keep track of vignetting effects (for all objects), then you could additionally record `image.added_flux` after the `drawImage` call.  That records how much flux actually made it onto the stamp when drawing.

Reply:
Thinking more about this, I realized that the other realized_flux wouldn't actually be right anymore given the way the code works here.  I think what we want here is to keep the previous realized_flux, but then don't re-draw the poisson variate here.  Rather, just apply the vignetting.
```
flux = self.realized_flux*self.vignetting.at_sky_coord(
                    base['sky_pos'], self.image.wcs, pix_to_fp)
self.realized_flux = round(self.realized_flux)
```

Reply:
I've implemented this.